app/src/main/java/com/duckduckgo/app/bookmarks/model/BookmarksRepository.kt

```diff
@@ -82,6 +87,10 @@ class BookmarksDataRepository(
         return bookmarksDao.getBookmarksByParentIdSync(parentId)
     }
 
+    override suspend fun getBookmarkFolderByParentId(parentId: Long): BookmarkFolder? {
+        return bookmarkFoldersDao.getBookmarkFolders().firstOrNull()?.find { it.id == parentId }
```

Comment:
Haven't checked, but I think filtering at SQL level is more efficient than getting the whole list of folders here, and then filtering. Can you create a new method in `BookmarkFoldersDao`, and just use it here?

Reply:
Created a new method please check.